File: src/main/scala/pw/ian/spigotjs/SpigotJS.scala
package pw.ian.spigotjs

import java.io.InputStreamReader
import javax.script.ScriptEngineManager

import org.bukkit.plugin.java.JavaPlugin

class SpigotJS extends JavaPlugin {

  override def onEnable() = {
    val engine = new ScriptEngineManager().getEngineByName("nashorn")
    engine.eval(new InputStreamReader(this.getResource("jvm-npm.js")))
    engine.eval("require('./scripts')")
  }

}

File: src/main/java/pw/ian/spigotjs/SpigotJS.java
package pw.ian.spigotjs;

import java.io.InputStreamReader;
import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;
import javax.script.ScriptException;

import org.bukkit.plugin.java.JavaPlugin;

public class SpigotJS extends JavaPlugin {

  @Override
  public void onEnable() {
    ScriptEngine engine = new ScriptEngineManager().getEngineByName("nashorn");
    try {
      engine.eval(new InputStreamReader(this.getResource("jvm-npm.js")));
      engine.eval("require('./scripts')");
    } catch (ScriptException ex) {
      getLogger().severe("Could not load scripts!");
      ex.printStackTrace();
    }
  }

}
